cmake/external/CudaRdcUtils.cmake

```diff
@@ -613,24 +642,46 @@ function(cuda_rdc_depends_on OUTVARNAME lib potentialdepend)
       get_target_property(lib_link_libraries ${lib} LINK_LIBRARIES)
       get_target_property(lib_interface_link_libraries ${lib} INTERFACE_LINK_LIBRARIES)
       if(lib_interface_link_libraries)
-        list(APPEND lib_link_libraries lib_interface_link_libraries)
-        list(REMOVE_DUPLICATES lib_link_libraries)
+        if(lib_link_libraries)
+	   list(APPEND lib_link_libraries ${lib_interface_link_libraries})
+        else()
+	   set(lib_link_libraries ${lib_interface_link_libraries})
+	endif()
```

Comment:
This looks like an optimization; does it actually help in practice?

Reply:
Which part?  I replace:
```
        if(lib_link_libraries)
	   list(APPEND lib_link_libraries ${lib_interface_link_libraries})
```
with simply
```
	   list(APPEND lib_link_libraries ${lib_interface_link_libraries})
```
I ended up having to deal with a library named `lib_link_libraries-NOTFOUND`

Reply:
ahhhh that's subtle. I was thinking it was empty. Maybe a little comment would be good? (I wonder if a macro `append_if_found` would slow it down?)

Reply:
I think I will just add a comment.  The `append_if_found` used here would be more aptly named `append_if_list_sofar_is_invalid_replace_if_not`